salt/utils/http.py

```diff
@@ -567,11 +567,13 @@ def query(url,
         except tornado.httpclient.HTTPError as exc:
             ret['status'] = exc.code
             ret['error'] = six.text_type(exc)
+            log.error("Cannot perform 'http.query': {0} - {1}".format(url_full, ret['error']))
```

Comment:
I'd avoid logging these, because you do return the errors. Passing an invalid URI is expected too.

Reply:
Thanks for your review @rares-pop, IMHO it does make sense to add this `log.error` message here. There are few reasons for that:

- It keeps consistency with the rest of the `http.query` method, where every time an error is produce, we return with `ret['error']` and also log the error in the minion log:

https://github.com/saltstack/salt/blob/310a30e8648ccc52d5bc91f0cda5e477821cbae3/salt/utils/http.py#L207-L208
https://github.com/saltstack/salt/blob/310a30e8648ccc52d5bc91f0cda5e477821cbae3/salt/utils/http.py#L525-L526
https://github.com/saltstack/salt/blob/310a30e8648ccc52d5bc91f0cda5e477821cbae3/salt/utils/http.py#L677-L678

- It also makes this situation easier for debugging since the error message is kept on the minion logs and not only displayed on the return (which can be lost or covered in case you're not directly calling this method, i.a. as part of Jinja inside of an SLS file).

Reply:
I am also in favor of consistency in this case.

Reply:
@meaksh, @xeacott - this is an execution module. If you pass invalid parameters, it will return an error.

Logging is for when something internally went wrong.

How is that logging useful in any way? You don't even have to debug that code, because it returns the error immediately. You only have to check the status of the function.

To be consistent you would have to remove all the logs from there, or at least downgrade them to log.debug but I fail to see any meaning in those.

Reply:
I agree with @rares-pop here - if you get a 400 or 403 or 500, presumably each one of those will show up in this log message. That's not terribly useful to dump to a log.

Reply:
Thanks for the discussion here. @rares-pop @waynew I see your point here, and I agree that when we get a HTTP Error from the remote hosts, then it would be ok not logging it.

However, IMHO I would keep the `log.error` when a `socket` exception is raised since, on that case, the HTTP request is not even executed due some error. Then I consider useful to produce an error message that reaches the minion log files.

I'm removing logging for the `HTTPError` exception.

Reply:
Yes, the HTTP request is not executed in that case, but the function will return saying it wasn't executed and why. I still think it's pointless to log it. Logging is useful when you don't know what happens, to avoid silent fails. But here you do. The function will return the exact error you want to duplicate in the log file.

Reply:
@rares-pop, what happens when you're calling this inside a Jinja? For example, I have an SLS file which does something like:

```
{% if salt['http.query'](my_url, status=True).get('status') == 200 %}
...
{% endif %}
```

In this case, if something happens on that `http.query` execution and we don't log anything, it will not be easy to know what happened. Neither looking at the minion logs.

Reply:
I'm not an expert on jinja, but that doesn't look very good.

You should be doing more like:
`ret = salt['http.query'](my_url, status=True)`
`if (ret.get('status') == 200)`:
`   ...`
`else`
`  log the error % ret`

The function is returning everything you need. If you choose to cherry-pick certain info, then you will loose details. But you shouldn't rely on the log to tell you what you ignored.

If I'm the only one who see things in this way, please disregard my comment.

Reply:
After looking back into this, I do agree with @rares-pop here. The `return` should have everything useful about what happened during the call and its given back to you.